app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments, through: :posts
+  has_many :likes, through: :posts
```

Comment:
Please make sure to fix this association and remove the through method because this isn't a many to many relationship, it's just a user has many comments and each comment only belongs to this user.
```ruby
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```